src/js/dqa.jsx

```diff
@@ -21,122 +15,212 @@ var SortTypes = {
         return {
+            columnWidths: {
+                code: width,
+                desc: 3*width,
+                status: 2*width,
+                sites: 3*width,
+            },
             isCollapsed: false,
-            rows: this._getRows(),
-            sortBy: null,
-            sortDir: null,
+            rows: {},
+            sortBy: 'code',
+            sortDir: SortTypes.ASC,
+            tableWidth: 500,
         };
     },
 
-    _getRows: function() {
+    _getRows: function(props, sortBy) {
         var code;
+        var desc;
         var rows = [];
         var status;
-        var numSites;
+        var sites;
+
+        if (!sortBy) {
+            sortBy = this.state.sortBy;
+        }
+
+        for (code in props.data) {
+            desc = props.dict[code]? props.dict[code].desc : '';
+
+            for (status in props.data[code]) {
+                if (this.props.aggregated) {
+                    sites = [];
+                    for (site in props.data[code][status]) {
+                        sites.push(site + '(' + props.data[code][status][site].length + ')');
+                    }
+                } else {
+                    sites = props.data[code][status].map(function(obj) {
+                        return obj.site;
+                    });
+                }
 
-        for(code in this.props.data) {
-            for(status in this.props.data[code]) {
-                numSites = this.props.data[code][status].length;
-                rows.push([code, '', status, numSites]);
+                // if the table is sorted by site, list sites individually;
+                // if the table is sorted by anything other than site, 
+                // keep sites grouped by (code + status)
+                if (sortBy === 'sites') {
+                    for (i in sites) {
+                        rows.push({
+                            code: code,
+                            desc: desc,
+                            status: status,
+                            sites: sites[i],
+                        });
+                    }    
+                } else {
+                    sites = sites.reduce(function(a,b) {
+                        return a + ', ' + b;
+                    });
+
+                    rows.push({
+                        code: code,
+                        desc: desc,
+                        status: status,
+                        sites: sites,
+                    });
+                }
             }
         }
         return rows;
     },
 
+    componentWillReceiveProps: function(nextProps) {
+        this.setState({
+            rows: this._getRows(nextProps),
+        })
+    },
+
+    componentWillMount: function() {
+        this.setState({
+            rows: this._getRows(this.props),
+        })
+    },
+
     componentDidMount: function() {
-        this._sortRowsBy(0);
+        // set a timeout for updating width, otherwise everything on the page gets
```

Comment:
This block is repeated code (update the component on a window resize). Move to a utility function?